### ISSUE TITLE
JDK8 images only published for first patch release 2.346.1 of final release cycle 

### ISSUE THREAD

**Issue by awaddell** [jenkins-infra/helpdesk#3099]
### Service(s)

Docker Hub

### Summary

Like many JDK8 laggards, we have deployed the 'final JDK8 release' to production.

We would now like to deploy a later patch release for the many fixes therein.

However, there has been no 2.346.2 or 2.346.3 docker image published because a change was merged, IMO prematurely, on 25 June [Removing Java 8 by basil · Pull Request #1356 · jenkinsci/docker · GitHub](https://github.com/jenkinsci/docker/pull/1356)

Expectation are that the final JDK8 release means the release cycle which, IIRC, is 3 months for LTS releases.

Can JDK8 please be supported for the full release cycle (or at least the latest patch - 2.346.3) ?

**Comment by dduportal** [comment]
Many thanks for reporting this @awaddell ! Good catch that these JDK8 declination are missing. Gotta fix this today.

**Comment by dduportal** [comment]
cc @NotMyFault @timja for information

- There is a branch, in the jenkinsci/docker repository, named `stable-2.346` which goal is to manage this case (until the next LTS liefcylce is started) and we forgot to run container builds from this branch.
- I'm currently dry-running this branch on trusted.ci.jenkins.io (private controller used to deploy official container images) to see what it should deploy if run. Goal is to make it build and deploy the missing jdk8 images.

**Comment by dduportal** [comment]
- Since the `master` branch publication published the `jdk17` tags for 2.346.2 and 2.346.3, I'll let the `stable-2.346` branch build to publish the missing `jdk17-preview` tags. Rationale: a end user using the [lts-jdk17-preview](https://hub.docker.com/layers/jenkins/jenkins/jenkins/lts-jdk17-preview/images/sha256-12b7fdf0fa412470c8ba65e13b1c243624d53a023c3d4d197edc3cd006c709fe?context=explore) tag should have the version 2.346.3 of Jenkins.
  - Please note that the `-preview` suffix was removed for weekly and will be for the next LTS line of course, keeping the `jdk17`.
- I'll cherry-pick, in a PR, the commit https://github.com/jenkinsci/docker/commit/db736b6eaa3e39987e47b8bb04882cfd0e41ce9a to bump the jenkins plugin CLI for the missing images (to stay coherent with the already published LTS images)

**Comment by NotMyFault** [comment]
Thanks for the heads up!

> and we forgot to run container builds from this branch.

Assuming this is typically done when the release lead requests to run the job on trusted.ci?

**Comment by timja** [comment]
> Assuming this is typically done when the release lead requests to run the job on trusted.ci?

well it's done now, it was a temporary thing and we've finished that line.
we just got about it after the .1

**Comment by dduportal** [comment]
> 

That's it. It's why I'm gonna open a PR to the `stable-2.346` branch to only build what is missing (and adjust eventual dependencies). This branch is aimed at being temp until next LTS.

**Comment by basil** [comment]
A minor side effect is that the 2.346.3 release included an update from OpenJDK 11.0.15 to 11.0.16, which also required an upgrade of Pipeline: Groovy. I think at least one user was surprised by this so I opened https://github.com/jenkins-infra/jenkins.io/pull/5371 to clarify the upgrade guide.

I realize that opinions vary, but my opinion is that the packaging of a given LTS line (including both OS packaging and Docker packaging) should be branched and changes should be backported individually. While that may be a little more work for maintainers, I think it benefits users by adhering to the principle of least surprise and forcing maintainers to justify and document backports as they are committed.

I don't feel strongly about this, since I realize that the status quo in the Docker packaging is to use one branch for everything. It is up to the maintainers to make decisions like this, regardless of my opinion.

**Comment by dduportal** [comment]
> While that may be a little more work for maintainers

It's not only a matter of work but also how much does it rely on human specific behavior ("knowledge of git, knowledge of domain boundaries, how people are handling these changes and a lot more of subjectives elements"). 

WDYT about an automated process for opening the change requests for each change merged into the principal branch in such a branching model? That would shift the maintainer effort from "don't forget to pick this change" to "should we pick this change" which would be the right question to ask (with answers depending on context).

I also wonder about the trade-off between "least surprise" vs. "bump JDK from latest to get the bugfix and seuciryt fix" for instance?

**Comment by basil** [comment]
Yeah, for packaging repositories where there is very little feature development and where we expect most changes to be safe bug fixes or dependency updates, automatic backport PRs could be very helpful indeed, especially if they take away the pain of running Git commands and opening pull requests. The backports could be reviewed by the release lead/officer who should have overall context about the content of the next stable release. If a changelog is needed, the docs team could be pinged in the corresponding backport PR. And if a given backport turns out to have been too aggressive, it can always be conservatively reverted toward the next stable release (e.g., if a dependency update backported to 2.361.1 was too aggressive, it could be conservatively reverted for 2.361.2 to keep the line stable, while remaining in the weekly release for bleeding edge users to work through the problems).

**Comment by dduportal** [comment]
Please maintainers, I need a review of https://github.com/jenkinsci/docker/pull/1439 to unblock these missing images

**Comment by dduportal** [comment]
Thanks folks!

The JDK8 images had been published with success.
